ctoec/data-collection
# Add error messaging to upload

## Background
At some point, we had error messaging that informed a user of an incorrectly formatted file upload. That messaging disappeared, so this PR adds it back in while also simplifying the messaging states/cases to always direct the user to upload data only in one of our templates.

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1355

## Validation Plan
- Open and log into app
- Go to file upload page
- [ ] Try to upload a csv or xlsx that has different formatting from the template
  - [ ] Verify error message appears and directs user to the template
  - [ ] Error alert disappears on page refresh or successful file upload
- [ ] Try to upload a file of the completely wrong type (i.e. a .zip, .jpg, etc.)
  - [ ] Verify error message appears and directs user to the template
  - [ ] Error alert disappears on page refresh or successful file upload

## Automated Testing
- [ ] All e2e tests are passing.

## Additional Context
The reason for testing on two different file types is that we have two different error cases in the back end depending on the type of error that happens while parsing. Spreadsheet files with improperly formatted columns will trigger an `ApiError` version of a 400, (lines 62-66 in `src/routes/enrollmentReport.ts`), while files that aren't spreadsheets at all will trigger the more general `BadRequestError` type of 400 (lines 67-69). We want this error message to appear for both of these cases, since the ticket's intention in simplifying the messaging was to handle all 400s by having the user use a proper template.
